test/thread_pool.cpp

```diff
@@ -22,4 +24,28 @@ TEST(ThreadPool, should_execute_function_in_another_thread) {
   EXPECT_NE(tid, std::this_thread::get_id());
 }
 
+TEST(ThreadPool, processes_all_queued_tasks_when_destroyed) {
+  static constexpr size_t TASK_COUNT = 1;
+
+  pc::static_thread_pool pool{1};
+
+  std::atomic<bool> can_start{false};
+  std::atomic<size_t> processed_tasks_count{0};
+
+  for (size_t i = 0; i < TASK_COUNT; ++i) {
+    post(pool.executor(), [&can_start, &processed_tasks_count] {
+      while (!can_start)
```

Comment:
rest of the tests use `latch` for synchronization like this. [For example](https://github.com/VestniK/portable_concurrency/blob/d6c388f928a6dd9c22467a190fdc5cfb7090abd4/test/notify.cpp#L75-L81)